### PR TITLE
feat(addon): aws cloudwatch metrics helm addon

### PR DESCRIPTION
**Pull request by askulkarni2**
### What does this PR do?

Fixes https://github.com/aws-ia/terraform-ssp-amazon-eks/issues/35

### Motivation

With this we addon support for Container Insights.

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [X] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [X] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [X] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [X] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [X] E2E Test successfully complete before merge?

### CODE REVIEW

File: docs/add-ons/aws-cloudwatch-metrics.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Container Insights collects data as performance log events using embedded metric format. These performance log events are entries that use a structured JSON schema that enables high-cardinality data to be ingested and stored at scale. From this data, CloudWatch creates aggregated metrics at the cluster, node, pod, task, and service level as CloudWatch metrics. The metrics that Container Insights collects are available in CloudWatch automatic dashboards, and also viewable in the Metrics section of the CloudWatch console.
+
+## Usage
+
+[aws-cloudwatch-metrics](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/modules/kubernetes-addons/aws-cloudwatch-metrics) can be deployed by enabling the add-on via the following.
+
+```hcl
+enable_aws_for_fluentbit = true
```
Comment (bryantbiggs) [inline docs/add-ons/aws-cloudwatch-metrics.md:12]:
```suggestion
enable_aws_cloudwatch_metrics = true
```

File: modules/kubernetes-addons/aws-cloudwatch-metrics/main.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module "helm_addon" {
+  source            = "../helm-addon"
+  manage_via_gitops = var.manage_via_gitops
+  set_values        = local.set_values
+  helm_config       = local.helm_config
+  irsa_config       = local.irsa_config
+  addon_context     = var.addon_context
+}
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/aws-cloudwatch-metrics/main.tf:8]: ending new line

File: modules/kubernetes-addons/aws-cloudwatch-metrics/values.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+clusterName: ${eks_cluster_id}
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/aws-cloudwatch-metrics/values.yaml:1]: ending new line

File: modules/kubernetes-addons/aws-cloudwatch-metrics/locals.tf [hunk 4]
```diff
@@ -0,0 +1,48 @@
+locals {
+  name                 = "aws-cloudwatch-metrics"
+  namespace            = "amazon-cloudwatch"
+  service_account_name = "cloudwatch-agent"
+
+  set_values = [
```
Comment (kcoleman731) [inline modules/kubernetes-addons/aws-cloudwatch-metrics/locals.tf:6]: most locals have this prop towards the bottom. We should try and keep ordering consistent.

File: docs/add-ons/aws-cloudwatch-metrics.md (hunk 1 above)
Comment (askulkarni2) [reply docs/add-ons/aws-cloudwatch-metrics.md]: done

File: modules/kubernetes-addons/aws-cloudwatch-metrics/values.yaml (hunk 3 above)
Comment (askulkarni2) [reply modules/kubernetes-addons/aws-cloudwatch-metrics/values.yaml]: done

**Review by kcoleman731 — COMMENTED** [review]
Looking good!

**Review by bryantbiggs — APPROVED** [review]

File: modules/kubernetes-addons/aws-cloudwatch-metrics/locals.tf (hunk 4 above)
Comment (askulkarni2) [reply modules/kubernetes-addons/aws-cloudwatch-metrics/locals.tf]: addressed

File: modules/kubernetes-addons/aws-cloudwatch-metrics/main.tf (hunk 2 above)
Comment (askulkarni2) [reply modules/kubernetes-addons/aws-cloudwatch-metrics/main.tf]: done

**Review by vara-bonthu — APPROVED** [review]
Nice work @askulkarni2 👍🏼  LGTM

**Pull request merged**